packages/@internationalized/number/src/NumberParser.ts

```diff
@@ -119,23 +126,50 @@ class NumberParserImpl {
       .replace(this.symbols.minusSign, '-')
       .replace(this.symbols.numeral, this.symbols.index);
 
+    if (this.options.style === 'percent') {
+      // javascript is bad at dividing by 100 and maintaining the same significant figures, so perform it on the string before parsing
+      let isNegative = fullySanitizedValue.indexOf('-');
+      fullySanitizedValue = fullySanitizedValue.replace('-', '');
+      let index = fullySanitizedValue.indexOf('.');
+      if (index === -1) {
+        index = fullySanitizedValue.length;
+      }
+      fullySanitizedValue = fullySanitizedValue.replace('.', '');
+      if (index - 2 === 0) {
+        fullySanitizedValue = `0.${fullySanitizedValue}`;
+      } else if (index - 2 === -1) {
+        fullySanitizedValue = `0.0${fullySanitizedValue}`;
+      } else if (index - 2 === -2) {
+        fullySanitizedValue = '0.00';
+      } else {
+        fullySanitizedValue = `${fullySanitizedValue.slice(0, index - 2)}.${fullySanitizedValue.slice(index - 2)}`;
+      }
+      if (isNegative > -1) {
+        fullySanitizedValue = `-${fullySanitizedValue}`;
+      }
+    }
+
     let newValue = fullySanitizedValue ? +fullySanitizedValue : NaN;
     if (isNaN(newValue)) {
       return NaN;
     }
 
+    if (this.options.style === 'percent') {
+      // extra step for rounding percents to what our formatter would output
+      let options = {
+        ...this.options,
+        style: 'decimal',
+        minimumFractionDigits: Math.min(this.options.minimumFractionDigits + 2, 20),
+        maximumFractionDigits: Math.min(this.options.maximumFractionDigits + 2, 20)
+      };
+      return (new NumberParser(this.locale, options)).parse(new NumberFormatter(this.locale, options).format(newValue));
```

Comment:
Should this rounding be specific to percents? Seems like it would need to apply to other styles as well?

Reply:
It should be applied to others as well. I haven't figured out a good way to do that yet though. This works for percents because it sets the style option to decimal, instead of percent, so we can never end up back in this section of code causing an infinite loop.

I don't know currently how to do that more generically and prevent the infinite recursive loop.

I have a test for it
```
    describe('fraction digits', function () {
      it('should round for maximum fraction digits', function () {
        expect(new NumberParser('en-US', {maximumFractionDigits: 2}).parse('10.555')).toBe(10.56);
      });
    });
```